app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatabaseService.kt

```diff
@@ -309,6 +311,22 @@ class FirestoreDatabaseService internal constructor(
         return documentRef.get().await().toObject(UserSchema::class.java)!!.toPublic()
     }
 
+    //TODO Optimize transaction to db
+    override suspend fun setAdminAccess(firebaseUser: FirebaseUser, adminAccess: Boolean): User {
+        val uid = firebaseUser.uid
+        val type = User.Type.FIREBASE
+        val documentRef = users.document("$uid@$type")
+        val data = UserSchema().apply {
+            isAdmin = adminAccess
+        }
+        documentRef.set(data).await()
+        return documentRef.get().await().toObject(UserSchema::class.java)!!.toPublic()
+    }
+
+    override suspend fun checkIsAdmin(fireStoreUser: User): Boolean {
```

Comment:
Would this method not be more useful if it took a user id as an argument?

Reply:
For me No , but we have to ask @NicolasVial who uses the backend for the UI separation flow

Reply:
I think it is easy like this, we can get the user with the presenter  using authPresenter.currentUser.